internal/parse/parse.go

```diff
@@ -0,0 +1,378 @@
+package parse
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	internalUtil "github.com/containers/buildah/internal/util"
+	"github.com/containers/common/pkg/parse"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage"
+	"github.com/containers/storage/pkg/idtools"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	selinux "github.com/opencontainers/selinux/go-selinux"
+	"github.com/pkg/errors"
+)
+
+const (
+	// TypeBind is the type for mounting host dir
+	TypeBind = "bind"
+	// TypeTmpfs is the type for mounting tmpfs
+	TypeTmpfs = "tmpfs"
+	// TypeCache is the type for mounting a common persistent cache from host
+	TypeCache = "cache"
+	// mount=type=cache must create a persistent directory on host so its available for all consecutive builds.
+	// Lifecycle of following directory will be inherited from how host machine treats temporary directory
+	BuildahCacheDir = "buildah-cache"
+)
+
+var (
+	errBadMntOption = errors.New("invalid mount option")
+	errBadOptionArg = errors.New("must provide an argument for option")
+	errBadVolDest   = errors.New("must set volume destination")
+)
+
+// GetBindMount parses a single bind mount entry from the --mount flag.
+// Returns specifiedMount and a string containing value of `from` if `from` was used with mount.
```

Comment:
Please update this to mention that the string value is the image, if we mounted one, and empty otherwise.  Returning the image's ID would still be better for finding exactly what to unmount later, for cases where the name gets assigned to a different image while the bind mount is in use.